### PR TITLE
Layout component

### PR DESCRIPTION
**Pull request by luucvanderzee**
- Created `vgg-grid` component, that can be used to set `vgg-section`s into a grid layout with no particular order, just filling out the space
- Created `vgg-repeat` component, that can be used to create matrix-like grid layouts, where the x and y directions actually have meaning (like in a scatterplot matrix)
- Fixed 'binned' data axes
- Mapping object syntax now also working for deeply nested children of `vgg-map` component (was first limited to direct children). BEWARE: doesn't work if the child is within another scoped slot!

### CODE REVIEW

File: docs/core/repeat.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+title: 'Repeat'
+---
+
+# Component tag
+
+`<vgg-repeat>`
+
+# Description
+
+The Grid component is used to quickly position and align [Section](./section.md)
```
Comment (GraceGSy) [inline docs/core/repeat.md]: The Repeat component?

**Comment by GraceGSy** [comment]
Hey, this looks really cool! (and definitely makes the scatterplot matrix easier than whatever I'm doing now.)

A couple of minor questions:

- For the repeat component, can I add an additional variable (other than `x` and `y`)? For example, is it possible for me to scale the scatterplot `fill` by a third variable?
- For the grids, is there a way to skip a cell/leave a cell empty if needed?

File: docs/core/repeat.md (hunk 1 above)
Comment (luucvanderzee) [reply docs/core/repeat.md]: Good find!

**Comment by luucvanderzee** [comment]
@GraceGSy 
>     * For the repeat component, can I add an additional variable (other than `x` and `y`)? For example, is it possible for me to scale the scatterplot `fill` by a third variable?

Depends on what you mean. The x and y props correspond to the rows and columns of the matrix. The number of rows or columns will be calculated based on the number of entries that the Array passed to `:x` or `:y` has. But there are no rules about what you can put in this array. So you could do something like

```
<vgg-repeat v-slot="{ x, y }" :x="[ { j: 'foo', k: 'bar' }, { j: 'baz', k: 'qux' } ... ]">

  <vgg-map v-slot="{ row }">

    <vgg-point 
      :x="{ val: row[x.j] }"
      :fill="{ val: row[x.k] }" 
    />

  </vgg-map>

</vgg-repeat>
```

So in this way you can have more complex repetitions of graphics, but each column will still have the same values for `x.j` and `x.k`. I don't know if we could add more props... how would they behave? I am not sure

>     * For the grids, is there a way to skip a cell/leave a cell empty if needed?

If you want to leave out some cells, you can just use `v-if`! If we modify the `RepeatLayout.vue` example to exclude the matrix cells that are plotting the same variable on both the x and y axis:

```
<vgg-repeat
  v-slot="{ x, y }"
  :x="['a', 'b', 'c']"
  :y="['a', 'b', 'c']"
  :cell-padding="20"
>

  <vgg-section v-if="x !== y">

    <vgg-map v-slot="{ row }">

      <vgg-point
        :x="{ val: row[x], scale: x }"
        :y="{ val: row[y], scale: y }"
        :fill="{ val: row[x], scale: x }"
      />

    </vgg-map>

  </vgg-section>

</vgg-repeat>
```

you get

screenshot from 2019-02-21 10-22-21

**Comment by bianchi-dy** [comment]
This is quite timely! Is it possible to change the dimensions of each 'cell' in the grid progressively, i.e. dependent on the row/column of the cell? Like so: 
screenshot 2019-02-21 at 11 34 07 pm

**Comment by luucvanderzee** [comment]
@bianchi-dy I am not sure if the performance will even improve that much. The major reason that the performance for drawing rectangles is currently quite slow is because of [this issue](https://github.com/spatialnetworkslab/vue-gg/issues/56). The first optimization listed in that issue is actually fairly straightforward to implement, if you want I can give this priority and fix that tomorrow before I continue with my other todo lists?

Other than that, it could be cool to have grid cells of different sizes depending on their contents, but that would be a lot more difficult and might be kind of a custom thing that you can solve with your own code (exactly like you are doing already!). But if you don't think it is as one-off/custom as I think, we can consider adding it in the future

**Comment by bianchi-dy** [comment]
@luucvanderzee for the rectangle rendering we're going to generate all the combinations of graphs (number of data points vs. number of dimensions) in the next two weeks so yes, that would be helpful.

As for the changing cell size, it's most likely a custom thing so we can ignore that for now until it comes up again.

**Pull request merged**